make/build_test.mk

```diff
@@ -39,14 +40,14 @@ $($(T)_TEST_OBJ_DIR)/%.o: $($(T)_TEST_ROOT)/%.c | $(T) $(dir $($(T)_TEST_OBJ))
 # Compile the test runners
 $($(T)_TEST_OBJ_DIR)/%.o: $($(T)_GEN_DIR)/%.c | $(T) $(dir $($(T)_TEST_RUNNERS_OBJ))
 	@echo "T: $(notdir $<) -> $(notdir $@)"
-	@$(CC) -MD -MP -w -c -o $@ $< $($(firstword $|)_CFLAGS) $(addprefix -I,$(INC_DIRS))
+	@$(CC) -MD -MP -w -c -o $@ $($(T)_GEN_DIR)/$(notdir $<) $($(firstword $|)_CFLAGS) $(addprefix -I,$(INC_DIRS))
```

Comment:
This won't work - you're trying to `$(T)` within a rule. You also shouldn't need to make this change.